thrift/arg2/kv_iterator.go

```diff
@@ -0,0 +1,88 @@
+// Package arg2 contains tchannel thrift Arg2 interfaces for external use.
+//
+// These interfaces are currently unstable, and aren't covered by the API
+// backwards-compatibility guarantee.
+package arg2
+
+import "encoding/binary"
+
+// KeyValIterator is a iterator for reading tchannel-thrift Arg2 Scheme,
+// which has key/value pairs (k~2 v~2).
+// NOTE: to be optimized for performance, we try to limit the allocation
+// done in the process of iteration.
+type KeyValIterator struct {
+	arg2Payload           []byte
+	leftPairCount         int
+	keyOffset             int
+	valueOffset, valueLen int
+}
+
+// InitKeyValIterator inits a KeyValIterator with the buffer pointing at
+// start of Arg2.
+// NOTE: tchannel-thrift Arg Scheme starts with number of key/value pair.
+func InitKeyValIterator(arg2Payload []byte) (KeyValIterator, bool) {
```

Comment:
why `Init` instead of `NewKeyValueIterator`?

Reply:
Initially it's operated on a struct, but since we are changing it to return a struct, I will update the name to NewKeyValIterator.